docs/user-flows.md

```diff
@@ -0,0 +1,247 @@
+# User Flows in Lighthouse
+
+> Motivation: You want to run Lighthouse on your whole site, not just the landing page.
+
+> Is that checkout form after you add an item to your cart accessible?
+
+> What is the Cumulative Layout Shift of my SPA page transition?
+
+You want Lighthouse on a _flow_, not just a page load.
+
+This document describes how Lighthouse measures flows and offers recommendations on how to structure your own flow measurement with practical examples.
+
+## Flow Building Blocks
+
+Flow measurement in Lighthouse is enabled by three modes: navigations, timespans, and snapshots. Each mode has its own unique use cases, benefits, and limitations. Later, you'll create a flow by combining these three core report types.
+
+### Navigation
+
+Navigation reports analyze a single page load. Navigation is the most common type of report you'll see. In fact, all Lighthouse reports prior to v9 are navigation reports.
+
+#### Benefits
+
+- Provides an overall performance score and all metrics.
+- Contains the most advice of all report types (both time-based and state-based audits are available).
+
+#### Limitations
+
+- Cannot analyze form submissions or single page app transitions.
+- Cannot analyze user interaction performance.
+- Cannot analyze content that isn't available immediately on page load.
+
+#### Use Cases
+
+- Obtain a Lighthouse Performance score.
+- Measure First Contentful Paint, Largest Contentful Paint, Speed Index, or Time to Interactive.
+- Assess Progressive Web App capabilities.
+
+#### Code
+
+```js
+import {writeFileSync} from 'fs';
+import puppeteer from 'puppeteer';
+import lighthouse from 'lighthouse';
+
+async function main() {
+  const browser = await puppeteer.launch();
+  const page = await browser.newPage();
+  const flow = new lighthouse.UserFlow(page);
```

Comment:
nit: should we expose a `newUserFlow`? perhaps exposing the api via a class isn't necessary / ideal. idk

Reply:
cc @adamraine thoughts?

Reply:
The ES6 class made the most sense to me, is the benefit here that it matches the puppeteer methods?

Reply:
I think @connorjclark is referring to the *public* entrypoint being something non-constructor based, not that we should move away from the class itself.

I have to agree constructor-based public APIs feel gross to me for a node library, but admittedly I don't have a compelling case for why, so I'm fine with either.

Reply:
Aight I slept on this one and a function entry point SGTM.

Reply:
see #13139